lambda/rotator-snowflake/rotate/databricks.go

```diff
@@ -0,0 +1,100 @@
+package rotate
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/chanzuckerberg/go-misc/lambda/rotator-snowflake/setup"
+	"github.com/chanzuckerberg/go-misc/snowflake"
+	"github.com/pkg/errors"
+	"github.com/xinsnake/databricks-sdk-golang/aws/models"
+)
+
+type snowflakeUserCredentials struct {
+	user            string
+	role            string
+	pem_private_key string
+}
+
+func (creds *snowflakeUserCredentials) writeSecrets(secretsClient setup.SecretsIface, currentScope string) error {
+	if currentScope == "" {
+		return errors.New("empty scope")
+	}
+
+	err := secretsClient.PutSecret([]byte(creds.user), currentScope, "snowflake.user")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to put secret username %s in scope %s", creds.user, currentScope)
+	}
+
+	err = secretsClient.PutSecret([]byte(creds.role), currentScope, "snowflake.role")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to put role %s in scope %s", creds.role, currentScope)
+	}
+
+	err = secretsClient.PutSecret([]byte(creds.pem_private_key), currentScope, "snowflake.pem_private_key")
+	return errors.Wrapf(err, "Unable to put private key %s in scope %s", creds.pem_private_key, currentScope)
```

Comment:
this might log the private key somewhere. We might want to avoid leaking that info. How do you feel about omitting it from the error?

Reply:
ohh, I didn't think of that.

Reply:
ok i'm gonna omit it

Reply:
done